src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -248,7 +253,7 @@ public ScrollQueryFetchSearchResult executeScan(InternalScrollSearchRequest requ
         contextProcessing(context);
         try {
             processScroll(request, context);
-            if (context.searchType() == SearchType.COUNT) {
+            if (context.searchType() == SearchType.QUERY_THEN_FETCH) {
```

Comment:
don't we have to set `context.size(0)` as well here? In general (not related to the change) - why do we need this flip flop between search type?